### PR TITLE
Add probe source part to kubernetes cmd properties of ChaosEngine manifest

### PR DESCRIPTION
**Pull request by rogeriofbrito**
## Proposed changes

Fixes the evaluation probe issue that occurs when source mode is enabled (probe executing inside a container). More details can be found in bellow referenced issue.
Fix: https://github.com/litmuschaos/litmus/issues/4693

Probe executing in a k6 container after fix:
Screenshot 2024-09-12 at 16 52 08

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

**Comment by rogeriofbrito** [comment]
The backend-checks pipeline step have indicated a problem with imports order in file `./chaoscenter/upgrade-agents/control-plane/versions/v0.0.0/manager.go`, but this file wasn't modificated in this PR. How to proced in this case?

**Comment by rogeriofbrito** [comment]
@namkyu1999 I have updated the fix branch with the master branch. Could you approve the workflow to run the pipeline?

File: chaoscenter/graphql/server/pkg/probe/utils/utils.go [hunk 1]
```diff
@@ -347,6 +347,12 @@ func GenerateProbeManifest(probe *model.Probe, mode model.Mode) (string, error)
 			_probe.RunProperties.StopOnFailure = *probe.KubernetesCMDProperties.StopOnFailure
 		}
 
+		if probe.KubernetesCMDProperties.Source != nil {
+			var source v1alpha1.SourceDetails
+			_ = json.Unmarshal([]byte(*probe.KubernetesCMDProperties.Source), &source)
```
Comment (S-ayanide) [inline chaoscenter/graphql/server/pkg/probe/utils/utils.go:352]: We should log the err if any on this unmarshal

**Review by S-ayanide — CHANGES_REQUESTED** [review]

File: chaoscenter/graphql/server/pkg/probe/utils/utils.go (hunk 1 above)
Comment (S-ayanide) [reply chaoscenter/graphql/server/pkg/probe/utils/utils.go:352]: Please also verify if you are able to run a CMD probe without providing any source input at all

Comment (rogeriofbrito) [reply chaoscenter/graphql/server/pkg/probe/utils/utils.go]: > We should log the err if any on this unmarshal

I have added a warn log (without return the error) when any fail ocurr when unmarshalling the source part of the probe. Thus, the source property of struct CmdProbeInputs will remain nil and in practice the source part of the probe will be ignored.

> Please also verify if you are able to run a CMD probe without providing any source input at all

When we doesn't provide any source the evaluation of the probe works as expected:

Screenshot 2024-10-11 at 11 19 30

**Review by S-ayanide — APPROVED** [review]
LGTM

**Comment by Saranya-jena** [comment]
Hi @rogeriofbrito can you resolve the conflicts and check the DCO?

**Comment by rogeriofbrito** [comment]
@S-ayanide Hi, I placed all changes of this MR in a new single commit and force the push. In this new commit I have fixed all issues related with DCO and conflicts with master branch.

In [this](https://drive.google.com/file/d/1Pr63WNAkOFQVBbEop178j1ULbvoTYNQS/view?usp=sharing) screen recording is shown the probe with source part being evaluated in edge mode in a pod-delete experiment.

**Pull request merged**